src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,73 @@
+package duke.main;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.parse.Parser;
+import duke.store.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ *      File name: duke.main.Duke.java
+ *      @author: Jerome Neo
+ *      Description: The main class for the duke.main.Duke application.
+ */
+public class Duke {
+    private static Storage store = new Storage();
+    private static Ui display = new Ui();
+    private static Parser logic = new Parser();
+    private static TaskList tasks = new TaskList();
+
+    /**
+     * The main method.
+     * @param args
+     * @throws IOException
+     * @throws DukeException
+     */
+    public static void main(String[] args) throws IOException, DukeException {
+        display.greeting();
+        Scanner scanner = new Scanner(System.in);
+        String input;
+        while (true) {
+            input = scanner.nextLine();
+            if (!logic.isValidCommand(input)) {
+                display.somethingWentWrong();
+                break;
+            }
+
+            if (logic.isTaskCommand(input)) {
+                Task task = logic.toTask(input);
+                TaskList.addTask(task);
+                display.line();
+            } else if (input.equals("bye")) {
+                display.farewell();
+                break;
+            } else if (input.equals("list")) {
+                tasks.printTaskList();
+                display.line();
+            } else if (input.startsWith("mark")) {
+                int taskNumber = logic.indexToMark(input);
+                tasks.taskMarkedAtIndex(--taskNumber);
```

Comment:
I personally don't like using the prefix increment thing, because it's usually terribly confusing (increment then return? return then increment?)

Reply:
I agree with this. Decrementing it first and then using this would be much more clearer.